lib/iris/tests/integration/test_netcdf.py

```diff
@@ -903,5 +907,541 @@ def test_netcdf_with_no_constraint(self):
+            self.input_filepaths, self.result_filepath
+        )
+        return self.result_filepath
+
+    def check_expected_results(
+        self, global_attr_value=None, var_attr_vals=None
+    ):
+        """
+        Run checks on the generated output file.
+
+        The counterpart to create_testcase, with similar control arguments.
+        Check existence (or not) of : a global attribute, named variables, and their
+        local attributes.  Values of 'None' mean to check that the relevant global/local
+        attribute does *not* exist.
+        """
+        # N.B. there is only ever one result-file, but it can contain various variables
+        # which came from different input files.
+        ds = netCDF4.Dataset(self.result_filepath)
+        if global_attr_value is None:
+            assert self.attrname not in ds.ncattrs()
+        else:
+            assert self.attrname in ds.ncattrs()
+            assert ds.getncattr(self.attrname) == global_attr_value
+        if var_attr_vals:
+            var_attr_vals = self._default_vars_and_attrvalues(var_attr_vals)
+            for var_name, value in var_attr_vals.items():
+                assert var_name in ds.variables
+                v = ds.variables[var_name]
+                if value is None:
+                    assert self.attrname not in v.ncattrs()
+                else:
+                    assert self.attrname in v.ncattrs()
+                    assert v.getncattr(self.attrname) == value
+
+    #######################################################
+    # Tests on "user-style" attributes.
+    # This means any arbitrary attribute which a user might have added -- i.e. one with
+    # a name which is *not* recognised in the netCDF or CF conventions.
+    #
+
+    def test_usertype_single_global(self):
+        self.create_testcase(
+            attr_name="myname",  # A generic "user" attribute with no special handling
+            global_value_file1="single-value",
+            vars_values_file1={
+                "myvar": None
+            },  # the variable has no such attribute
+        )
+        # Default behaviour for a general global user-attribute.
+        # It simply remains global.
+        self.check_expected_results(
+            global_attr_value="single-value",  # local values eclipse the global ones
+            var_attr_vals={
+                "myvar": None
+            },  # the variable has no such attribute
+        )
+
+    def test_usertype_single_local(self):
+        # Default behaviour for a general local user-attribute.
+        # It results in a "promoted" global attribute.
+        self.create_testcase(
+            attr_name="myname",  # A generic "user" attribute with no special handling
+            vars_values_file1={"myvar": "single-value"},
+        )
+        self.check_expected_results(
+            global_attr_value="single-value",  # local values eclipse the global ones
+            # N.B. the output var has NO such attribute
```

Comment:
TODO: from the comment, I think this is missing a `var_attr_vals={"myvar": None}`